### PR TITLE
refactor: event route, enhance code and error handling

### PR DESCRIPTION
**Pull request by kychok98**
### Refactor backend event route
Description: enhance code readability, add more error handling

#### Checklist
- [ ] I've followed the Contributing guidelines provided in the repository.
- [ ] I've made the changes which were demanded in the linked issue.
- [ ] I've tested my code on a Chromium based browser.
- [ ] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

### CODE REVIEW

File: backend/controllers/eventController.js
Comment (anktrj) [inline backend/controllers/eventController.js]: There are some breaking changes! Before merging the PR changes needs to be addressed on front-end!
After the required changes has been done on the client, this can be merged!
BTW
Great work! 
LGTM

**Comment by ankiiitraj** [comment]
@all-contributors please add @kychok98 for code

File: backend/controllers/eventController.js
Comment (EmperorYP7) [inline backend/controllers/eventController.js]: Sort the events according to `eventDate` instead of `createdAt`.

File: backend/routes/event.js
Comment (EmperorYP7) [inline backend/routes/event.js:8]: Add separate getters for `/all` where we retrive all the events and `/` where we retrieve only the first 5 when sorted in decreasing order according to `eventDate`.

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]
I think we need one more route to be managed before we merge this. This is a great implementation though. Although, I prefer using promises instead of try-catch for asynchronous functions. Great to have your contribution! :smile:

**Comment by kychok98** [comment]
> I think we need one more route to be managed before we merge this. This is a great implementation though. Although, I prefer using promises instead of try-catch for asynchronous functions. Great to have your contribution! 😄

I think using await the code will be much more concise and clean, when codebase getting large, .then will make the code harder to read. It's a good practice to learn await 
ref: https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

File: backend/routes/event.js
Comment (kychok98) [reply backend/routes/event.js:8]: I have go through the preview code, where the event route for `/` and `/all` are trying to achieve the same result, differences for both them are just adding the `.limit(5`), where it violated the design principle DRY, 
So for current `router.get("/", eventController.get_event); `route,
we can pass the req.body.options, json object
`{ options: { limit: 5} }`
to achieve the same result 

*FYI in future for the frontend event table, it can make a paginate feature using this route options too

Comment (anktrj) [reply backend/routes/event.js:8]: That's correct, I got to learn about DRY today. Thanks!

**Comment by ankiiitraj** [comment]
> > I think we need one more route to be managed before we merge this. This is a great implementation though. Although, I prefer using promises instead of try-catch for asynchronous functions. Great to have your contribution! smile
> 
> I think using await the code will be much more concise and clean, when codebase getting large, .then will make the code harder to read. It's a good practice to learn await
> ref: https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

That's correct, promise chaining is an issue. I was hesitant to learn about async-await. Now I think I got it!
Will merge it after front-end get compatible!

**Review by tusharjain0022 — APPROVED** [review]
I think this PR is now ready to get merged 🚀. @EmperorYP7, @ankiiitraj?

**Pull request merged**
